### PR TITLE
[TIMOB-26032] iOS/Android: Fix Ti.Contacts parity, deprecate old API's properly

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26032

`no tests` because it only refactors existing code. Also requires UI-related interaction (contacts permissions, selecting values from the modal dialog).

### CODE REVIEW

File: android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsModule.java [hunk 1]
```diff
@@ -136,6 +136,13 @@ public void save(Object people)
 
 	@Kroll.method
 	public PersonProxy getPersonByID(long id)
+	{
+		Log.w(TAG, "getPersonByID has been deprecated in favor of getPersonById (parity)", Log.DEBUG_MODE);
```
Comment (sgtcoolguy) [inline android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsModule.java]:
```java
Log.w(TAG, "Ti.Contacts.getPersonByID() has been deprecated in favor of Ti.Contacts.getPersonByIdentifier() for cross-platform parity", Log.DEBUG_MODE);
```

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
The deprecation note on Android has a type in the correct method name. Added suggested fixed text.

File: android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsModule.java (hunk 1 above)
Comment (hansemannn) [reply android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsModule.java]: Updated!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.
Note - @hansemannn There seems linting issue in Contacts.java as build is failing. Can you fix that.

**Comment by lokeshchdhry** [comment]
FR Passed.

**Pull request merged**